### PR TITLE
Add ability to plot HPD credible interval in 1D histograms

### PR DESCRIPTION
**Pull request by soumide1102**
Rebooting the HPD credible interval PR #2331 here. That PR is out of date and might not be compatible with the current PyCBC Inference code. Now made `--plot-marginal` take a string input. Choices are `hpd` and `percentiles`, and defaults to `None` like  @cmbiwer  suggested. There's a `--marginal-hpd-percent` option to give the code the HPD % CI to plot. Kept that separate from `--marginal-percentiles` that reads in the input for `plot-marginal=percentiles`, as using a common option for reading the percentiles for the `percentiles` method and % CI for the HPD method could be confusing.

If `plot-marginal=percentiles`: will plot the percentiles provided in `--marginal-percentiles`. If `--marginal-percentiles` is not provided will plot the 5, 50, 95 percentiles.

If `plot-marginal=hpd`: will plot the `--marginal-hpd-percent` % HPD CI. If `--marginal-hpd-percent` is not provided, will plot the 90% HPD CI.

### CODE REVIEW

File: pycbc/inference/option_utils.py [hunk 1]
```diff
@@ -262,13 +262,23 @@ def add_plot_posterior_option_group(parser):
     """
     pgroup = parser.add_argument_group("Options for what plots to create and "
                                        "their formats.")
-    pgroup.add_argument('--plot-marginal', action='store_true', default=False,
+    pgroup.add_argument('--plot-marginal', default=None, type=str,
```
Comment (cmbiwer) [inline pycbc/inference/option_utils.py]: You should consider adding a ``choices=["percentiles", "hpd"]`` here.

Comment (soumide1102) [reply pycbc/inference/option_utils.py]: Done.

**Comment by soumide1102** [comment]
@cdcapano Example of unimodal distribution, percentiles, 90% CI: https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/m1_percentiles.png
unimodal distribution, HPD, 90% CI  : https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/m1_hpd.png

Bimodal distribution, percentiles, 90% CI: https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/iota_percentiles.png
Bimodal distribution, HPD, 90% CI:  https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/iota_hpd.png

Bimodal distribution, percentiles, 50% CI: https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/iota_percentiles_50.png
Bimodal distribution, HPD 50% CI:  https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/iota_hpd_50.png

So the 50% HPD CI only includes the denser peak. So this code will not be able to handle disjoint multimodal regions. Will have to look at how that can be done.

**Comment by spxiwh** [comment]
I'm going to close this now as it's been inactive for quite some time. It's easy to reopen pull requests if needed.

**Pull request closed without merging**